parcels/collection/collectionaos.py

```diff
@@ -484,6 +533,206 @@ def add_same(self, same_class):
                 p._cptr = pdata
         self._ncount = self._data.shape[0]
 
+    def add_multiple(self, data_array):
+        """
+        Add multiple particles from an array-like structure (i.e. list or tuple or np.ndarray)
+        to the collection.
+        :arg data_array: one of the following:
+            i) a list or tuples containing multple Particle instances
+            ii) a Numpy.ndarray of dtype = Particle dtype
+            iii) a dict of Numpy.ndarray of shape, each of which with N = # particles
+        :returns vector-list of indices of all added particles
+        """
+        super().add_multiple(data_array)
+        results = []
+        if len(data_array) <= 0:
+            return results
+        if isinstance(data_array, list) or isinstance(data_array, tuple):
+            logger.info("SoA -> data_array - type: {}; values: {}".format(type(data_array[0]), data_array))
+            for item in data_array:
+                results.append(self.add_single(item))
+        elif isinstance(data_array, np.ndarray) and (data_array.dtype == self._ptype):
+            for i in range(data_array.shape[0]):
+                pdata = data_array[i]
+                results.append(self.add_single(pdata))
+        elif isinstance(data_array, dict) and isinstance(data_array['lon'], np.ndarray):
+            pu_ids = None
+            pu_indices = None
+            n_pu_data = 0
+            if MPI and MPI.COMM_WORLD.Get_size() > 1:
+                mpi_comm = MPI.COMM_WORLD
+                mpi_size = mpi_comm.Get_size()
+                mpi_rank = mpi_comm.Get_rank()
+                spdata = np.array([data_array['lon'], data_array['lat']]).transpose([1, 0])
+                min_pu = None
+                if mpi_rank == 0:
+                    dists = distance.cdist(spdata, self._pu_centers)
+                    min_pu = np.argmax(dists, axis=1)
+                    self._pu_indicators = np.concatenate((self._pu_indicators, min_pu), axis=0)
+                min_pu = mpi_comm.bcast(min_pu, root=0)
+                self._pu_indicators = mpi_comm.bcast(self._pu_indicators, root=0)
+                pu_indices = np.nonzero(min_pu == mpi_rank)[0]
+                ids = np.arange(ScipyParticle.lastID, stop=ScipyParticle.lastID+data_array['lon'].shape[0]) if 'id' not in data_array.keys() else data_array['id']
```

Comment:
it does not enforce the use of `Scipy`, as the `ScipyParticle.lastID` is equally the variable used by `JITParticle` to manage its current `lastID`. `lastID` is of type `Variable`, meaning that the variable itself is expressed as Numpy value or by the ctype storage in the background, flexing according to the derived particle class that is instanced in the end.

Regarding the second question: the ID needs to be globally valid, meaning: for any type of particle used within a simulation. In a context where one has 2 or more particle sets of different particles that are simultaneously computed, no ID is allowed to occur double. Therefore, one shall not modify the `lastID` value of the specific particle type here (as it may globally differ from other particles), but of the base particle type declaring the variable, which is `ScipyParticle`. I hope this in comprehensible - takes a bit to think of the possible, complicated class inheritance associations in this case, admittedly.

Reply:
Hmm, I still think it may be confusing for a future developer to see this here. If `ScipyParticle.lastID` is the same as `JITParticle.lastID`, then why not use the one associated with the type used? Is there no quick and simple way to generalise this?